python/pyiceberg/schema.py

```diff
@@ -800,7 +800,11 @@ def primitive(self, primitive: PrimitiveType) -> PrimitiveType:
 
 def prune_columns(schema: Schema, selected: Set[int], select_full_types: bool = True) -> Schema:
     result = visit(schema.as_struct(), _PruneColumnsVisitor(selected, select_full_types))
-    return Schema(*(result or StructType()).fields, schema_id=schema.schema_id, identifier_field_ids=schema.identifier_field_ids)
+    return Schema(
+        *(result or StructType()).fields,
+        schema_id=schema.schema_id,
+        identifier_field_ids=list(selected.intersection(schema.identifier_field_ids)),
```

Comment:
I don't think `intersection` is correct. The identifier field IDs are the set of fields that make a unique key. The resulting schema either has the same identifier fields (i.e. the same key) or doesn't have a key at all. I don't think it should ever have a different identifier/key.

So I think this should be `schema.identifier_field_ids if all(id in selected for id in schema.identifier_field_ids) else []`.